lib/iris/pandas.py

```diff
@@ -167,6 +167,19 @@ def as_data_frame(cube, dropna=True, asmultiindex=False):
     index = pandas.MultiIndex.from_product(coords, names=coord_names)
     data_frame = pandas.DataFrame({cube.name(): data.flatten()}, index=index)
 
+    # Add aux coord information
+    if add_aux_coord:
+        aux_coord_names = list(map(lambda x: x.name(), cube.aux_coords))
+        for acoord in add_aux_coord:
+            assert acoord in aux_coord_names, f'\"{acoord}\" not in cube' # Check aux coord exists
```

Comment:
Are `assert` statements too harsh?  Should an `AssertionError` be raised if `as_data_frame` can't find user specified `AuxCoords` or attributes in the cube, or pass silently and return a `DataFrame` but without the missing coords/attributes?

Reply:
Last week we discussed providing the basics but leaving it to the user to handle the more nuanced bits. In that vein I think the best course would be to add ALL the `AuxCoord`s - the vast majority of use cases would presumably be covered by either all or none, and those few who want some-not-all can easily remove unwanted columns downstream.

Reply:
Also I read last year that `assert` is only intended for things that you ALWAYS expect to be true - they're primarily a developer convenience. Anything that's there to inform the user should be in an `if:` block, raising the most appropriate `Error` class for that case.